servicetalk-concurrent-api/src/main/java/io/servicetalk/concurrent/api/DelayedCancellableThenSubscription.java

```diff
@@ -65,11 +63,11 @@ final void delayedCancellable(Cancellable delayedCancellable) {
     }
 
     final void delayedSubscription(Subscription subscription) {
-        // The operation corresponding to the first cancellable is considered done, so we dereference the Cancellable.
-        // Best effort is OK as subsequent calls to cancel should be NOOPs [1][2].
-        // [1] https://github.com/reactive-streams/reactive-streams-jvm#2.4
-        // [2] https://github.com/reactive-streams/reactive-streams-jvm#3.7
-        current = IGNORE_CANCEL;
-        delayedSubscription.delayedSubscription(subscription);
+        if (currentUpdater.getAndAccumulate(this, delayedSubscription,
```

Comment:
I'm not convinced we need the additional atomics on top of `DelayedSubscription`. IIUC the best effort semantics should be good enough, or does this cause a problem? This approach adds additional atomics on the non-cancel path which I assume would be more common (if we are completing work).